src/compressor/zlib/ZlibCompressor.cc

```
@@ -192,6 +204,14 @@ int ZlibCompressor::compress(const bufferlist &in, bufferlist &out, std::optiona
   if (qat_enabled)
     return qat_accel.compress(in, out, compressor_message);
 #endif
+#ifdef HAVE_UADK
+  if (uadk_enabled) {
+    int uadk_ret = uadk_accel.compress(in, out, compressor_message);
+    if (uadk_ret == 0) {
+      return uadk_ret;
```

Unlike ISAL and QAT, this falls through to other ways to compress on errors.  Why is that?  Can UADK be enabled but all of a sudden stop working?

(The same question goes for the two decomression methods.)

---

For initial intention here is that if UADK failed working somehow, compression/ decompression process will continue trying without UADK.

---

i think it's probably better to return the error directly so it's clear that something's broken. we don't want to get stuck in a situation where compression performs poorly because we're trying UADK and falling back every time

---

What do you think @idryomov ?

---

I agree with Casey that it's better to return the error, like it's done for other compressors.  That is why I commented in the first place ;)

---

NP, revised.